### ISSUE TITLE
Pkg should not hard crash like this

### ISSUE THREAD

**Issue by PetrKryslUCSD** [JuliaLang/Pkg.jl#2736]
The following error occurs:
```
PS E:\00WIP\FinEtoolsFlexStructures.jl> julia
  Activating environment at `E:\00WIP\FinEtoolsFlexStructures.jl\Project.toml`
ERROR: LoadError: TypeError: in typeassert, expected VersionNumber, got a value of type Pkg.Types.VersionSpecStacktrace:
 [1] load_urls(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec})
   @ Pkg.Operations C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\Operations.jl:545
 [2] #download_source#57
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\Operations.jl:732 [inlined]  
 [3] download_source
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\Operations.jl:731 [inlined]  
 [4] instantiate(ctx::Pkg.Types.Context; manifest::Nothing, update_registry::Bool, verbose::Bool, platform::Base.BinaryPlatforms.Platform, allow_build::Bool, allow_autoprecomp::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1406
 [5] instantiate
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1325 [inlined]        
 [6] #instantiate#252
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1321 [inlined]        
 [7] instantiate()
   @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1321
 [8] top-level scope
   @ C:\Users\PetrKrysl\Portable_Julia_VSCode\assets\.julia-1.6.2-depot\config\startup.jl:6
in expression starting at C:\Users\PetrKrysl\Portable_Julia_VSCode\assets\.julia-1.6.2-depot\config\startup.jl:4
PS E:\00WIP\FinEtoolsFlexStructures.jl>
```
I get that the project and the julia versions are mismatched, but a hard crash is a problem.
For instance, when that happens in vscode, the error message is lost and the user is left wondering
what went wrong.

**Comment by DilumAluthge** [comment]
In Julia 1.7 and later, manifest files include the Julia version, and using a manifest from a different Julia version will print an informative warning.

**Comment by IanButterworth** [comment]
And v1.6.2 upwards supports those new manifests, though it defaults to the old style. i.e. if you want to use the new manifest format with 1.6.3, create the manifest in 1.7 and then resolve it again in 1.6.3

**Comment by PetrKryslUCSD** [comment]
Agreed, 1.7 handles this gracefully, but with 1.6.2 I get the above hard crash.

**Comment by PetrKryslUCSD** [comment]
Given that 1.6 seems to be not going anywhere for a while, I think it should be fixed.

**Comment by DilumAluthge** [comment]
What Julia version generated the manifest in this example?

If it was Julia 1.5 or earlier, I don't think that there is anything we can do, because I think it's very unlikely that we'll make any more releases of Julia 1.5 or earlier.

**Comment by DilumAluthge** [comment]
Never mind, I see what you're getting at. Fix this in 1.6 by detecting the issue earlier and printing an informative error. Let me look into this.

**Comment by DilumAluthge** [comment]
#2737

**Comment by DilumAluthge** [comment]
> For instance, when that happens in vscode, the error message is lost and the user is left wondering
> what went wrong.

Off-topic, but this seems like a separate problem. When Julia or Pkg throws an error, VS Code should be able to show the error message to the user.

**Comment by PetrKryslUCSD** [comment]
> When Julia or Pkg throws an error, VS Code should be able to show the error message to the user.

Sure, I was just pointing out some of the consequences of a hard crash. :-)

**Comment by DilumAluthge** [comment]
Fixed in 1.6 by https://github.com/JuliaLang/Pkg.jl/pull/2737
